enumerable.rb

```diff
@@ -0,0 +1,117 @@
+module Enumerable
+  def my_each
+    return to_enum(:my_each) unless block_given?
+
+    each do |x|
+      yield x
+    end
+    self
+  end
+
+  def my_each_with_index
+    return to_enum __method__ unless block_given?
+
+    idx = 0
+    each do |x|
+      yield(x, idx)
+      idx += 1
+    end
+    self
+  end
+
+  def my_select
+    return to_enum unless block_given?
+
+    array = []
+    my_each do |x|
+      array.push(x) if yield(x) == true
+    end
+    array
+  end
+
+  def my_all?(reg = nill)
+    if !reg.nill?
+      my_each { |x| return false unless reg == x }
+    elsif block_given?
+      my_each { |x| return false unless yield x }
+    else
+      my_each { |x| return false unless x }
+    end
+    true
+  end
+
+  def my_any?(reg = nill)
+    res = false
+    if !reg.nill?
+      my_each { |x| res = true if reg == x }
+    elsif block_given?
+      my_each { |x| res = true if yield x }
+    else
+      my_each { |x| res = true if x }
+    end
+    res
+  end
+
+  def my_none?(reg = nil, &block)
+    !my_any?(reg, &block)
+  end
+
+  def my_count(num = nill?)
+    counter = 0
+
+    if num
+      my_each { |x| counter += 1 if num == x }
+      counter
+    elsif block_given?
+      my_each { |x| counter += 1 if yield(x) }
+      counter
+    else
+      num
+    end
+  end
+
+  def my_map(proc = nil)
+    return to_enum(:my_map) if !block_given? && proc.nil?
+
+    array = []
+    if !proc.nil?
+      my_each do |x|
+        array.push(proc.call(x))
+      end
+    else
+      my_each do |x|
+        array.push(yield(x))
+      end
+    end
+    arr
+  end
```

Comment:
#### my_map

- [ ] given a block returns a new array with the results of running the block once for every element in enum.

```ruby
puts [18, 22, 5, 6] .my_map {|num| num < 10 }  => [false, false, true, true]
```
- Real map retrieves the right answer
![image](https://user-images.githubusercontent.com/52209067/77464866-6d6ffb00-6dcd-11ea-9cc1-fefe86bcf840.png)

- Your my_map retrieves a NameError
![image](https://user-images.githubusercontent.com/52209067/77464928-88db0600-6dcd-11ea-8e45-d2b0885ef12f.png)

- [ ] enumerables my_map accepts a proc as an argument and returns a new array with the results of running the proc once for every element in enum
Given:
```ruby
block = proc { |num| num < (2 + 7) / 2 }
```
The result should be
```ruby
puts  [22,4,3,42,43].my_map(&block) => [false, false, true, false, false]
```

- Real map retrieves the correct answer
![image](https://user-images.githubusercontent.com/52209067/77463288-fcc7df00-6dca-11ea-8c76-5770dbd5dee3.png)

- Your my_map is returning a NameError
![image](https://user-images.githubusercontent.com/52209067/77463326-10734580-6dcb-11ea-86ea-8614751d652f.png)

[OPTIONAL]

- [ ] In the specifications of the project in the [Odin project](https://www.theodinproject.com/courses/ruby-programming/lessons/advanced-building-blocks) # 12, you were asked to modify your my_map method to take either a proc or a block. And if both a proc and a block are given, only execute the proc. The real map doesn't have this feature, but you were asked to have it so I strongly recommend working on it

Given 
```ruby
my_proc = Proc.new {|num| num > 10 }
```

Result should be
```ruby
puts [18, 22, 5, 6] .my_map(my_proc) {|num| num < 10 }  => [true, true, false, false]
```